lib/rubocop/cop/style/multiline_list.rb

```diff
@@ -0,0 +1,167 @@
+      #         baz)
+      #       do_something
+      #     end
+      #
+      #     # ignored with EnableMethodParams
+      #     def method foo,
+      #         bar
+      #       do_something
+      #     end
+      #
+      #     # bad with EnableMethodArgs
+      #     method(foo, bar,
+      #       baz)
+      #
+      #     # good with EnableMeethodArgs
+      #     method(
+      #       foo, bar,
+      #       baz)
+      #
+      #     # ignored with EnableMethodArgs
+      #     method foo, bar,
+      #       baz
+      #
+      #     # bad with EnableHashElements
+      #     { a: 1,
+      #       b: 2}
+      #
+      #     # good with EnableHashElements
+      #     {
+      #       a: 1,
+      #       b: 2 }
+      #
+      #     # bad with EnableArrayElements
+      #     [ :a,
+      #       :b]
+      #
+      #     # good with EnableArrayElements
+      #     [
+      #       :a,
+      #       :b]
+      #
+      class MultilineList < Cop
+        include OnMethodDef
+
+        MSG = 'Add a line break before the first element of a multi-line list.'
+
+        def on_method_def(node, _method_name, args, _body)
+          return unless cop_config['EnableMethodParams']
+
+          check_method(node, args.to_a)
+        end
+
+        def on_send(node)
+          return unless cop_config['EnableMethodArgs']
+
+          _receiver, _name, *args = *node
+
+          check_method(node, args)
+        end
+
+        def on_array(node)
+          return unless cop_config['EnableArrayElements']
+
+          # Check if we are using an implicit array.
+          source = node.loc.expression.source_line[node.loc.column..-1].lstrip
+
+          return if !explicit_array?(source) &&
+                    node.parent.masgn_type? &&
+                    !assignment_on_same_line?(node)
+
+          check_children(node, node.children)
+        end
+
+        def on_hash(node)
+          return unless cop_config['EnableHashElements']
+
+          # Check if we are using an implicit hash.
+          if node.loc.begin && node.loc.begin.is?('{')
+            check_children(node, node.children)
+          else
+            return unless method_uses_parens?(node.parent, node)
+            check_children(node, node.children, node.parent)
+          end
+        end
+
+        def autocorrect(node)
+          ->(corrector) { corrector.insert_before(node.loc.expression, "\n") }
+        end
+
+        private
+
+        def check_method(node, children)
+          return if children.empty?
+
+          return unless method_uses_parens?(node, children.first)
+
+          check_children(node, children)
+        end
+
+        def explicit_array?(source)
+          ['[', '%w', '%W'].any? { |a| source.start_with?(a) }
+        end
+
+        def assignment_on_same_line?(node)
+          source = node.loc.expression.source_line[0...node.loc.column]
+          source =~ /\s*\=\s*$/
+        end
+
+        def method_uses_parens?(node, limit)
```

Comment:
This one is hard to rewrite so that it doesn't use regexp matching, because it handles both method definitions (two kinds) and method calls. So either leave it as it is, or break it up into two separate methods.